tests/foreman/cli/test_fact.py

```diff
@@ -61,3 +61,38 @@ def test_negative_list_by_name(module_target_sat):
     assert (
         module_target_sat.cli.Fact().list(options={'search': f'fact={gen_string("alpha")}'}) == []
     )
+
+
+@pytest.mark.no_containers
+def test_positive_update_client_facts(
+    module_target_sat, rhel8_contenthost, module_org, module_location, module_activation_key
+):
+    """Update client facts and verify the facts are updated on Satellite.
+
+    :id: ea94ccb7-a125-4be3-932a-bfcb035d3604
+
+    :steps:
+        1. Add another interface to the host.
+        2. Update the facts on the Satellite.
+        3. Verify that the facts are updated on the Satellite.
```

Comment:
Actually these steps seem we're only gathering facts. To update them there should be different order and more steps, e.g:

- register host and gather facts
- add interface
- update facts
- verify updated facts (verify that interface was added)

Reply:
Facts get updated during registration, so we would not need additional steps to update facts. Rest of the steps are added.

Reply:
Sometimes it misses updating facts so I have added an additional step.